pennylane/devices/tests/test_gates.py

```diff
@@ -33,6 +35,58 @@
 # ==========================================================
 # Some useful global variables
 
+# gates for which device support is tested
+ops = {
+    'BasisState': qml.BasisState(np.array([0]), wires=[0]),
+    'CNOT': qml.CNOT(wires=[0, 1]),
+    'CRX': qml.CRX(0, wires=[0, 1]),
+    'CRY': qml.CRY(0, wires=[0, 1]),
+    'CRZ': qml.CRZ(0, wires=[0, 1]),
+    'CRot': qml.CRot(0, 0, 0, wires=[0, 1]),
+    'CSWAP': qml.CSWAP(wires=[0, 1, 2]),
+    'CZ': qml.CZ(wires=[0, 1]),
+    'DiagonalQubitUnitary': qml.DiagonalQubitUnitary(np.eye(2), wires=[0]),
+    'Hadamard': qml.Hadamard(wires=[0]),
+    'MultiRZ': qml.MultiRZ(0, wires=[0]),
+    'PauliX': qml.PauliX(wires=[0]),
+    'PauliY': qml.PauliY(wires=[0]),
+    'PauliZ': qml.PauliZ(wires=[0]),
+    'PhaseShift': qml.PhaseShift(0, wires=[0]),
+    'QubitStateVector': qml.QubitStateVector(np.array([1., 0.]), wires=[0]),
+    'QubitUnitary': qml.QubitUnitary(np.eye(2), wires=[0]),
+    'RX': qml.RX(0, wires=[0]),
+    'RY': qml.RY(0, wires=[0]),
+    'RZ': qml.RZ(0, wires=[0]),
+    'Rot': qml.Rot(0, 0, 0, wires=[0]),
+    'S': qml.S(wires=[0]),
+    'SWAP': qml.SWAP(wires=[0, 1]),
+    'T': qml.T(wires=[0]),
+    'Toffoli': qml.Toffoli(wires=[0, 1, 2]),
+    # 'Beamsplitter': qml.Beamsplitter(0, 0, wires=[0, 1]),
+    # 'CatState': qml.CatState(0, 0, 0, wires=[0]),
+    # 'CoherentState': qml.CoherentState(0, 0, wires=[0]),
+    # 'ControlledAddition': qml.ControlledAddition(0, wires=[0, 1]),
+    # 'ControlledPhase': qml.ControlledPhase(0, wires=[0, 1]),
+    # 'CrossKerr': qml.CrossKerr(0, wires=[0, 1]),
+    # 'CubicPhase': qml.CubicPhase(0, wires=[0]),
+    # 'DisplacedSqueezedState': qml.DisplacedSqueezedState(0, 0, 0, 0, wires=[0]),
+    # 'Displacement': qml.Displacement(0, 0, wires=[0]),
+    # 'FockDensityMatrix': qml.FockDensityMatrix(np.array([0]), wires=[0]),
+    # 'FockState': qml.FockState(1, wires=[0]),
+    # 'FockStateVector': qml.FockStateVector(np.array([0]), wires=[0]),
+    # 'GaussianState': qml.GaussianState(np.array([0., 0.]), np.eye(2), wires=[0]),
+    # 'Interferometer': qml.Interferometer(np.eye(2), wires=[0, 1]),
+    # 'Kerr': qml.Kerr(0, wires=[0]),
+    # 'QuadraticPhase': qml.QuadraticPhase(0, wires=[0]),
+    # 'Rotation': qml.Rotation(0, wires=[0]),
+    # 'SqueezedState': qml.SqueezedState(0, 0, wires=[0]),
+    # 'Squeezing': qml.Squeezing(0, 0, wires=[0]),
+    # 'ThermalState': qml.ThermalState(0, wires=[0]),
+    # 'TwoModeSqueezing': qml.TwoModeSqueezing(0, 0, wires=[0, 1]),
+}
```

Comment:
Note: It is not terribly essential that this list is 100% maintained. If a gate is missing, we simply don't test here that a device claiming to implement it really implements it. 

A misrepresentation like that is likely to be reported by users.

Reply:
Also, I added CV gates, since the test suite will soon be expanded to cv.

Reply:
(Minor suggestion): Would it be worth leaving the CV gates out for now? Perhaps they might be separated into another file